subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ShortCircuitEmptyConfigurationResolver.java

```diff
@@ -58,7 +58,7 @@ public ShortCircuitEmptyConfigurationResolver(ConfigurationResolver delegate, Co
 
     @Override
     public void resolveBuildDependencies(ConfigurationInternal configuration, ResolverResults result) {
-        if (configuration.getAllDependencies().isEmpty()) {
+        if (configuration.getAllDependencies().isEmpty() && configuration.getAllDependencyConstraints().isEmpty()) {
```

Comment:
:x: If we have no dependencies (and only constraints) then can't we still short-circuit the resolution?

Reply:
Changed